src/datastructure.h

```diff
@@ -57,14 +62,18 @@ typedef struct {
 	int overTime[OVERTIME_SLOTS];
 	unsigned int numQueriesARP;
 	bool new;
+	sqlite3_stmt* whitelist_stmt;
+	sqlite3_stmt* gravity_stmt;
+	sqlite3_stmt* blacklist_stmt;
+	bool *regex_enabled[2];
 } clientsData;
 
 typedef struct {
 	unsigned char magic;
-	unsigned char regexmatch;
 	size_t domainpos;
 	int count;
 	int blockedcount;
+	ucharvec *clientstatus;
```

Comment:
This vector is not safe to be used by consumers of shared memory, as it contains pointers which point to the process's private memory.

Reply:
This should also trigger an updated shared memory version.

Reply:
This is intentional. Realizing this in FTL itself instead of in shared memory is a lot easier. Also, this is more of an internal DNS cache for FTL, nothing an external users should really look at. The contained information (whether a specific query is blocked or not and if, why) is still available as always in the shared queries data.